go-runtime/ftl/call.go

```diff
@@ -44,21 +43,61 @@ func Call[Req, Resp any](ctx context.Context, verb Verb[Req, Resp], req Req) (re
 	}
 }
 
-// Call a Sink through the FTL controller.
-// func CallSink[Req any](ctx context.Context, sink Sink[Req]) error {
-// }
+// Call a Verb through the FTL Controller.
+func Call[Req, Resp any](ctx context.Context, verb Verb[Req, Resp], req Req) (resp Resp, err error) {
+	callee := VerbToRef(verb)
+	return call[Req, Resp](ctx, callee.ToProto(), req)
+}
+
+// CallSink calls a Sink through the FTL controller.
+func CallSink[Req any](ctx context.Context, sink Sink[Req], req Req) error {
+	callee := SinkToRef(sink)
+	verbRef := &schemapb.VerbRef{Module: callee.Module, Name: callee.Name}
+	_, err := call[Req, Unit](ctx, verbRef, req)
+	return err
+}
+
+// CallSource calls a Source through the FTL controller.
+func CallSource[Resp any](ctx context.Context, source Source[Resp]) (Resp, error) {
+	callee := SourceToRef(source)
+	verbRef := &schemapb.VerbRef{Module: callee.Module, Name: callee.Name}
+	fmt.Printf("source ref2: %v\n", verbRef)
+	return call[Unit, Resp](ctx, verbRef, Unit{})
+}
+
+// CallEmpty calls a Verb with no request or response through the FTL controller.
+func CallEmpty(ctx context.Context, empty Empty) error {
+	callee := EmptyToRef(empty)
+	verbRef := &schemapb.VerbRef{Module: callee.Module, Name: callee.Name}
+
+	_, err := call[Unit, Unit](ctx, verbRef, Unit{})
+	return err
+}
 
 // VerbToRef returns the FTL reference for a Verb.
 func VerbToRef[Req, Resp any](verb Verb[Req, Resp]) VerbRef {
 	ref := runtime.FuncForPC(reflect.ValueOf(verb).Pointer()).Name()
 	return goRefToFTLRef(ref)
 }
 
+// SinkToRef returns the FTL reference for a Sink.
 func SinkToRef[Req any](sink Sink[Req]) SinkRef {
 	ref := runtime.FuncForPC(reflect.ValueOf(sink).Pointer()).Name()
 	return SinkRef(goRefToFTLRef(ref))
 }
 
+// SourceToRef returns the FTL reference for a Source.
+func SourceToRef[Resp any](source Source[Resp]) SourceRef {
+	ref := runtime.FuncForPC(reflect.ValueOf(source).Pointer()).Name()
+	return SourceRef(goRefToFTLRef(ref))
+}
+
+// EmptyToRef returns the FTL reference for an Empty.
+func EmptyToRef(empty Empty) VerbRef {
+	ref := runtime.FuncForPC(reflect.ValueOf(empty).Pointer()).Name()
+	return goRefToFTLRef(ref)
+}
```

Comment:
This stuff might change once we use a generic `Ref` everywhere. But this is working for now

Reply:
@worstell says she has a change for this in flight

Reply:
Awesome I can work with her to get her change and mine merged and updated here.